contracts/src/xchain.ts

```diff
@@ -405,6 +406,27 @@ export async function tokenDetailsFromProvider(addr:string, provider:JsonRpcProv
   }
 }
 
+const ERC165Abi = [
+  "function supportsInterface(bytes4) public view returns (bool)",
+];
+const ERC721InterfaceId: string = "0x80ac58cd";
+const ERC1155InterfaceId: string = "0xd9b67a26";
+
+export async function getNftContractType(addr:string, provider:JsonRpcProvider): Promise<string | undefined> {
```

Comment:
Not everything supports EIP-165.

Reply:
Yes, but according to the specs, both EUR-721 and ERC-1155 mandates support for EIP-165, so it should be OK rely on in, right?